lib/dal/src/change_set_pointer.rs

```diff
@@ -0,0 +1,125 @@
+//! The sequel to [`ChangeSets`](crate::ChangeSet). Coming to an SI instance near you!
+
+use std::sync::{Arc, Mutex};
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use si_data_pg::PgError;
+use telemetry::prelude::*;
+use thiserror::Error;
+use ulid::{Generator, Ulid};
+
+use crate::workspace_snapshot::WorkspaceSnapshotId;
+use crate::{pk, standard_model, DalContext, StandardModelError, Timestamp, TransactionsError};
+
+const FIND: &str = include_str!("queries/change_set_pointers/find.sql");
+
+#[remain::sorted]
+#[derive(Debug, Error)]
+pub enum ChangeSetPointerError {
+    #[error("ulid monotonic error: {0}")]
+    Monotonic(#[from] ulid::MonotonicError),
+    #[error("mutex error: {0}")]
+    Mutex(String),
+    #[error("pg error: {0}")]
+    Pg(#[from] PgError),
+    #[error("serde json error: {0}")]
+    SerdeJson(#[from] serde_json::Error),
+    #[error("standard model error: {0}")]
+    StandardModel(#[from] StandardModelError),
+    #[error("transactions error: {0}")]
+    Transactions(#[from] TransactionsError),
+}
+
+pub type ChangeSetPointerResult<T> = Result<T, ChangeSetPointerError>;
+
+pk!(ChangeSetPointerId);
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct ChangeSetPointer {
+    pub id: ChangeSetPointerId,
+    #[serde(flatten)]
+    pub timestamp: Timestamp,
+    #[serde(skip)]
+    pub generator: Arc<Mutex<Generator>>,
```

Comment:
This doesn't need to be marked default, or told which function to call to populate when deserializing from the DB?

Reply:
There's two things going on here I believe. First, `skip` attribute indicates that the field should be neither serialized nor deserialized **and** that serde will use `Default::default()` when deserializing. Second, I believe that since `Generator` implements `Default` you do not need to specify a function to be called.

Reply:
I had forgotten that skip implied default (didn't require explicitly marking to use default, or to mark how to default the value).

Reply:
Yep! It's a bit confusing truthfully because `skip` means "don't do anything" followed by "oh yeah it actually uses `Default::default()` when deserializing".